### PR TITLE
SUS 2017 TTJets HT binned 600+ and TTJets 1L from Tbar.

### PR DESCRIPTION
**Pull request by bth5032**
Added cards for TTJets HT binned (above 600 GeV) and TTJets to 1 lepton from TBar samples. These are from SUSY for 2017.

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_madspin_card.dat [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#*                                                          *
+#************************************************************
+#Some options (uncomment to apply)
+#
+#directory for gridpack mode
+set ms_dir ./madspingrid
+
+#initialization parameters
+ set Nevents_for_max_weigth 250 # number of events for the estimate of the max. weight
+# set BW_cut 15                # cut on how far the particle can be off-shell
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_madspin_card.dat]: please remove commented lines

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import model sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if relevant for speed
+#import model sm-lepton_masses_no_b_mass
+
+define p = p b b~
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat]: please remove proton and jet definitions

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_madspin_card.dat (hunk 1 above)
Comment (bth5032) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_madspin_card.dat]: Hi Marco, you mean remove just 

set BW_cut 15

Or all lines 16-18?

Also should this be done for all the new cards or just the tt1023l_1l_tbar_5f_ckm_LO_MLM sample?

**Comment by bth5032** [comment]
Should this be done just for the tt1023l_1l_tbar_5f_ckm_LO_MLM sample or all the cards (i.e. also the TTBar HT binned samples)?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_madspin_card.dat (hunk 1 above)
Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_madspin_card.dat]: only line 18

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import model sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if relevant for speed
+#import model sm-lepton_masses_no_b_mass
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat]: remove commented input commands

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-1200to2500_LO_MLM/tt0123j_5f_ckm_HT-1200to2500_LO_MLM_madspin_card.dat [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#*                                                          *
+#************************************************************
+#Some options (uncomment to apply)
+#
+#directory for gridpack mode
+set ms_dir ./madspingrid
+
+#initialization parameters
+ set Nevents_for_max_weigth 250 # number of events for the estimate of the max. weight
+# set BW_cut 15                # cut on how far the particle can be off-shell
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-1200to2500_LO_MLM/tt0123j_5f_ckm_HT-1200to2500_LO_MLM_madspin_card.dat]: remove also here commented commands

**Comment by bth5032** [comment]
Alright, comments have been addressed.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM_madspin_card.dat [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#to properly limit the number of concurrent processes for grid running
+set max_running_process 1
+#
+# specify the decay for the final state particles
+ decay t > w+ b, w+ > all all
+ decay t~ > w- b~, w- > all all
+ decay w+ > all all
+ decay w- > all all
+
+#define ell+ = e+ mu+ ta+
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM_madspin_card.dat]: can you please check all the commented definitions...

Comment (bth5032) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM_madspin_card.dat]: Hi Luca, what do you mean by check the definitions? You mean check whether they need to be uncommented?

Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM_madspin_card.dat]: no, I mean that ALL the definitions that are commented should be deleted

Comment (bth5032) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM/tt0123j_5f_ckm_HT-2500toInf_LO_MLM_madspin_card.dat]: Okay thank you for the clarification, it should be done now

File: bin/MadGraph5_aMCatNLO/cards/production/2017/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import model sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if relevant for speed
+#import model sm-lepton_masses_no_b_mass
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat]: there are still commented commands...

**Comment by bth5032** [comment]
Hi, that was for me, I thought you could ignore it easily. The repo now has no extra stuff.

Best

File: bin/MadGraph5_aMCatNLO/cards/production/2017/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat (hunk 6 above)
Comment (bth5032) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat]: Hi Luca, I am not sure why you see comments there. The file is clean: 

https://github.com/bth5032/genproductions/blob/master/bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat

Comment (bth5032) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/tt0123j_1l_tbar_5f_ckm_LO_MLM/tt0123j_1l_tbar_5f_ckm_LO_MLM_proc_card.dat]: Ahh I see, those are not the cards, I only want to submit the cards in the 13TeV subdirectory.

**Comment by bth5032** [comment]
Hi Luca, I am only trying to add the cards for 5 samples (in the title):

tt0123j_1l_tbar_5f_ckm_LO_MLM
tt0123j_5f_ckm_HT-600to800_LO_MLM
tt0123j_5f_ckm_HT-800to1200_LO_MLM
tt0123j_5f_ckm_HT-1200to2500_LO_MLM
tt0123j_5f_ckm_HT-2500toInf_LO_MLM

They all sit in /bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ 

I'm sorry if I am not doing this properly, but if it is possible for you to only take those cards, that would be great. Otherwise, if I need to make this repo up-to-date with the current cmssw HEAD, then could you please help me understand how to do that?

**Comment by perrozzi** [comment]
```
git fetch --all
git merge upstream/master
```
where upstream is pointing to https://github.com/cms-sw/genproductions
cfr https://help.github.com/articles/adding-a-remote/
let me know in case of problems

**Comment by perrozzi** [comment]
I see a lot of DY cards too, is this expected?

**Comment by bth5032** [comment]
No those cards were in cmssw-master when I made the pull request but have since been removed I believe

**Comment by perrozzi** [comment]
ouch, a mess happened. I think I know why, you have to rebase instead of merge (my bad...)
```
git reset --hard HEAD~1
git rebase upstream/master
```

**Comment by bth5032** [comment]
Hi, I can't push to my repo anymore after the reset command because the history is behind the history in the repo....

Can you explain to me how git is making that diff? My branch is perfectly up to date with the cmssw-genproductions repo right now except for the new stuff I have (and any changes made to cmssw's repo in the last hour or two).

When I make a pull request, time will pass, your repo changes as you merge in other pull requests... Is it on me to simply watch for every change you make and update my repo accordingly? Or is the pull request tied to the commit that was at HEAD when I made it? 

If that's the case, I don't understand what happened here? I literally pulled your repo, added only my cards, pushed to my repo and made the pull request... I don't understand how one is supposed to do this without making an unreasonable amount of work... Can I send someone the text files... Literally all I am trying to do is add some text files to this repo...

However, all the scripts that pull the old cards from the pre2017 tag do several commits along the way, so I think the reason things are so messed up is because of those scripts.... I don't have any explanation otherwise...

**Comment by perrozzi** [comment]
ok, yes maybe things got a bit messy... I will try to fetch your branch and clean it up, and let you know.
you can certainly also close this PR, use a clean branch and put there your files...

**Comment by bth5032** [comment]
Okay thank you... Again the folders are.

1. /bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_1l_tbar_5f_ckm_LO_MLM/
2. /bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT-600to800_LO_MLM
3. /bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT-800to1200_LO_MLM
4. /bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT-1200to2500_LO_MLM
5. /bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/tt0123j_5f_ckm_HT-2500toInf_LO_MLM/

Best,
-Bobak

**Comment by bth5032** [comment]
Hi, I don't think I am seeing these cards in the repo yet. Is there anything I can do to help?

**Comment by perrozzi** [comment]
can you please make another clean PR? sorry I couldn't follow up yet, it's faster if you do it, thanks

**Pull request closed without merging**